server/routerlicious/packages/lambdas/src/scribe/summaryReader.ts

```diff
@@ -67,8 +71,8 @@ export class SummaryReader implements ISummaryReader {
                     messages,
                     fromSummary: true,
                 };
-            } catch {
-                summaryReaderMetric.success(`Returning default summary`);
+            } catch (error: any) {
+                summaryReaderMetric.error(`Returning default summary`, error);
```

Comment:
@pradeepvairamani @tanviraumi I was wondering if it would be appropriate to make this metric "fail" in case of error, both here and below (when `enableWholeSummaryUpload` is `false`). My dilemma is: it is true that the first call to `readLastSummary()` might fail due to the way we implemented things (initial blank summary uploaded by Alfred does not include `.serviceProtocol` tree, for example). But marking the metric as success is tricky in cases like this one, where we uncovered a real bug. It might have been easier to catch this bug if we were marking the metric as failed, and also including the error/exception.

What do you think? Does it make sense to mark this metric as error? Or do you have any ideas about how to ignore the error the first time, but later on pay attention to those errors?

Reply:
Sure. That makes sense. I was wondering if we can explicitly check for serviceProtocol and mark the metric as success in that case. That would not capture the cases where serviceProtocol is missing due to an error but at least we can be more confident about the metric?

Reply:
Seconding Pradeep. I like the idea of explicitly checking if .serviceProtocol exists, rather than throwing an error on expected behavior

Reply:
That sounds like a good idea, but are there any other "expected error" scenarios? Or is the missing `.serviceProtocol` the only thing that would be ignorable? Also, later, if `.serviceProtocol` somehow is not present, we would ignore that. But it would be a real failure. Are we sure we want to ignore that?

Reply:
In other words, I was wondering if there is another way of detecting these errors, possibly outside of `SummaryReader`. I think `SummaryReader` is used in two situations:

1. in the lambda factory, when everything is starting for a document https://github.com/microsoft/FluidFramework/blob/e31df8309ad9a96eef50267f5b2a80303f24eea3/server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts#L88-L92

2. in the lambda itself, but only if "value.operation.term > this.term" https://github.com/microsoft/FluidFramework/blob/e31df8309ad9a96eef50267f5b2a80303f24eea3/server/routerlicious/packages/lambdas/src/scribe/lambda.ts#L116-L120

If I'm understanding it correctly, the ignorable "`.serviceProtocol` missing" error only applies to situation #1. So would it make sense to rearrange things so we have a different metric/analysis on each case (1. vs 2. above)?

Reply:
Scratch that. If Scribe restarts for some reason, I think we would go through situation 1. again, even though the document might already exist and therefore `.serviceProtocol` is expected. I think it might be a tricky problem to solve so maybe ignoring `.serviceProtocol` might be a sufficient approach for now. It is definitely not worse then it currently is, as we currently ignore everything

Reply:
The initial summary we upload only has ".protocol" (https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services-shared/src/storage.ts#L98). And the way we write summaries (bottom-up for git, single shot for antler) may prevent us from writing an incomplete summary. So I think it's safe to ignore deli, scribe, and logtail for the initial summary.

Later, if we find an issue with the check, we can change the initial upload code to write empty blobs for those as well.

Reply:
Great - I updated error handling in `SummaryReader` (both when `enableWholeSummaryUpload` is enabled and disabled) to handle the initial scenario.